src/components/molecules/Visualizer/hooks.ts

```diff
@@ -103,6 +104,45 @@ export default ({
   );
   dropRef(wrapperRef);
 
+  const [viewport, setViewport] = useState<Viewport>();
+
+  const viewportResizeObserver = useMemo(
+    () =>
+      new ResizeObserver(entries => {
+        const [entry] = entries;
+        let width: number | undefined;
+        let height: number | undefined;
+
+        if (entry.contentBoxSize) {
+          // Firefox(v69-91) implements `contentBoxSize` as a single content rect, rather than an array
+          const contentBoxSize = Array.isArray(entry.contentBoxSize)
+            ? entry.contentBoxSize[0]
+            : entry.contentBoxSize;
+          width = contentBoxSize.inlineSize;
+          height = contentBoxSize.blockSize;
+        } else if (entry.contentRect) {
+          width = entry.contentRect.width;
+          height = entry.contentRect.height;
+        }
+
+        setViewport({
+          width,
+          height,
+          isMobile: width === undefined ? undefined : width <= 768,
```

Comment:
define const for a magic number outside the hook: `const mobileWidth = 760;`

Reply:
Updated.